File: 9-palindrome-number/palindrome-number.cpp
class Solution {
public:
    bool isPalindrome(int x) {
        int temp = x;
        long long res = 0;

        while(temp > 0){
            int rem = temp % 10;
            res = res*10 + rem;

            temp /= 10;
        }

        if(x == res){
            return true;
        } else {
            return false;
        }
    }
};

File: 9-palindrome-number/palindrome-number.java
class Solution {
    public boolean isPalindrome(long x) {
        if(x < 0){
            return false;
        }

        long rev = 0;
        long temp = x;

        while(temp > 0){
            long r = temp % 10;
            rev = rev * 10 + r;

            temp /= 10;
        }

        if(rev == x){
            return true;
        } else {
            return false;
        }
    }
}